modules/apps/transfer/types/token.go

```diff
@@ -0,0 +1,61 @@
+package types
+
+import (
+	"strings"
+
+	errorsmod "cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	denominternal "github.com/cosmos/ibc-go/v8/modules/apps/transfer/internal/denom"
+)
+
+// Validate validates a token denomination and trace identifiers.
+func (t Token) Validate() error {
+	if err := sdk.ValidateDenom(t.Denom); err != nil {
```

Comment:
I don't think we want to do SDK validation on the denom. The reasoning for using `ibc/{hash}` was that other chains may use denominations which don't meet our chains validation (the validation function in SDK can be changed). I think mostly just check that the denom is not empty?

Reply:
the actual `t.Denom` field will never be an ibc denom in the format of `ibc/hash`, the sdk validation should be fine, in our code the we end up using the token denom to get the hash like this

```golang
fullDenomPath := token.GetFullDenomPath()
trace := types.ParseDenomTrace(fullDenomPath)
coin := sdk.NewCoin(trace.IBCDenom(), 100)
```

Reply:
The issue isn't `ibc/hash`, the issue is that chainA's denom validation might be different than chainB. So if chainA wants to use an emoji token for its denom, chainB shouldn't fail validation on this (that's why chainB converts the emoji to a hash)

Reply:
https://github.com/cosmos/ibc-go/pull/6364